hardware/src/main/java/at/amartinz/hardware/utils/HwIoUtils.java

```diff
@@ -213,6 +213,40 @@ public static boolean fileExists(final String[] files) {
         return ((content != null) ? content.trim() : null);
     }
 
+	/**
+     * Read one line from a file with root credentials.
+     * Use blocking reading.
+     * @param Path of the file to read
+     * @return Line read.
+     */
+    @WorkerThread @Nullable public static String readOneLineRoot(final String path) {
+        final String content = readFileInternalRoot(path, true);
+        return ((content != null) ? content.trim() : null);
+    }
+
+	/**
+	 * Read file with root permissions.
+     * @param path Path of the file to read
+     * @param oneLine read only one line if true
+     * @return content read
+     */
+    @WorkerThread
+    @Nullable
+    public static String readFileInternalRoot(@Nullable final String path, boolean oneLine) {
+        if (TextUtils.isEmpty(path) || !RootCheck.isRooted()) {
+            return null;
+        }
+        String output = null;
+        if(oneLine)
+            output = RootShell.fireAndBlockStringNewline(
+                    new Command(String.format("head -n 1 %s", path)));
```

Comment:
Good question, worked for me so I assumed it was part of busybox but I'm not sure, I'll make some research.